### PR TITLE
fix(cdk: click-outside): fix click-outside bug

### LINKED ISSUE #245: [cdk:ClickOutside] 同时存在多个指令时，点击其中一个指令，不能触发其他指令的事件。
### Reproduction link

```html
<template>
  <ix-button @click.stop="toggle">toggle</ix-button>
  <p v-if="visible" v-click-outside="toggle" @click="log">Hello</p>
  <p v-if="visible" v-click-outside="toggle" @click="log">World</p>
</template>
<script lang="ts">
import { defineComponent, ref } from 'vue'
import { clickOutside } from '@idux/cdk/click-outside'

export default defineComponent({
  directives: { clickOutside },
  setup() {
    const visible = ref(false)

    const toggle = () => {
      visible.value = !visible.value
    }

    const log = () => {
      console.log('hello world')
    }

    return { visible, toggle, log }
  },
})
</script>

```
### Steps to reproduce

点击 `Hello` 或者点击 `World`

### What is expected?

点击 `Hello` , `World` 消失，反之亦然
### What is actually happening?

无论点击哪一个，另一个都没有消失
### Any additional comments? (optional)

---
| Environment | Info |
|---|---|
| @idux |     |
| Browser |     |

### PR DESCRIPTION
**Pull request by LaamGinghong**
fix #245

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IduxFE/idux/blob/main/docs/contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

**Comment by LaamGinghong** [comment]
我重写了测试用例，你看看

File: packages/cdk/click-outside/__tests__/clickOutside.spec.ts [hunk 1]
```diff
@@ -1,70 +1,72 @@
-import { computed, ref } from 'vue'
 import { mount } from '@vue/test-utils'
+import { IxButton } from '@idux/components/button'
```
Comment (danranVm) [inline packages/cdk/click-outside/__tests__/clickOutside.spec.ts:2]: 不要用 ix-button 吧，就普通的 button 不就完了，你这样会让 cdk 的 test 依赖 components...
我前两天给你删了的，你又给我整回来了。

demo 里 可以用。

File: packages/cdk/click-outside/docs/Index.zh.md [hunk 2]
```diff
@@ -19,5 +19,12 @@ cover:
 #### Props
 
 ```typescript
-type ClickOutside = (...args: unknown[]) => void
+interface ClickOutsideOptions {
+  exclude: Ref<HTMLElement>[]
+  handler: ClickOutsideHandler
+}
+
+type ClickOutsideHandler = () => void
```
Comment (danranVm) [inline packages/cdk/click-outside/docs/Index.zh.md:27]:
```suggestion
type ClickOutsideHandler = (e: Event) => void
```

File: packages/cdk/click-outside/src/useClickOutside.ts [hunk 3]
```diff
@@ -1,36 +1,61 @@
...
+
+on(document, 'click', event => {
+  documentHandlerMap.forEach(({ excludes, handler }, el) => {
+    const target = event.target as Node
+    const isContain = el.contains(target)
+    const isTargetExclude =
+      excludes.length && (excludes.some(item => item.contains(target)) || excludes.includes(target as HTMLElement))
+    const isSelf = target === el
+    if (isContain || isTargetExclude || isSelf) {
+      return
```
Comment (danranVm) [inline packages/cdk/click-outside/src/useClickOutside.ts:29]: 29 和 42-43 行，单测没覆盖，不好覆盖吗？

File: packages/cdk/click-outside/src/useClickOutside.ts [hunk 4]
```diff
@@ -1,36 +1,61 @@
...
+  exclude: Ref<HTMLElement>[]
+  handler: ClickOutsideHandler
+}
+
+type ClickOutsideHandler = () => void
+
+export type ClickOutsideBinding = ClickOutsideHandler | ClickOutsideOptions
+
+interface DocumentHandlerOptions {
+  excludes: HTMLElement[]
```
Comment (danranVm) [inline packages/cdk/click-outside/src/useClickOutside.ts:15]:
```suggestion
  exclude: HTMLElement[]
```

不用带 s 把，没见过带 s 的。。。

File: packages/cdk/utils/typeof.ts [hunk 5]
```diff
@@ -45,7 +45,7 @@ export function isPlainObject(val: unknown): val is Record<any, any> {
   return _toString.call(val) === '[object Object]'
 }
 
-export function isObject(val: unknown): val is Record<any, any> {
+export function isObject<T = Record<any, any>>(val: unknown): val is T {
```
Comment (danranVm) [inline packages/cdk/utils/typeof.ts:48]: 这个改回去，我之前也是这么写的，有个地方的 ts 推导会报错。

**Review by danranVm — CHANGES_REQUESTED** [review]

File: packages/cdk/click-outside/src/useClickOutside.ts (hunk 3 above)
Comment (LaamGinghong) [reply packages/cdk/click-outside/src/useClickOutside.ts:29]: 先忽略

**Pull request merged**